pyquil/api/_qam.py

```diff
@@ -60,18 +61,27 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: int = None,
+                     value: Optional[Union[int, float, Sequence[int], Sequence[float]]] = None):
         """
-        Writes a value into a memory region on the QAM at a specified offset.
+        Writes a value or chronologically unwraps a list of values into a memory region on
+        the QAM at a specified offset.
 
         :param region_name: Name of the declared memory region on the QAM.
         :param offset: Integer offset into the memory region to write to.
-        :param value: Value to store at the indicated location.
+        :param value: Value(s) to store at the indicated location.
         """
         assert self.status in ['loaded', 'done']
 
-        aref = ParameterAref(name=region_name, index=offset)
-        self._variables_shim[aref] = value
+        if isinstance(value, Sequence):
+            if offset != None:
```

Comment:
```suggestion
            if offset is not None:
```

Not sure what's going on, but this comment was made and resolved in the previous PR as well: https://github.com/rigetti/pyquil/pull/1079#discussion_r349224904

Ditto the comment about `Optional[int]` type for `offset`, above.  Probably worth reviewing all the comments from the previous PR #1079 to make sure none of them got dropped.